tokio/src/runtime/builder.rs

```diff
@@ -915,6 +923,33 @@ impl Builder {
             blocking_pool,
         ))
     }
+
+    #[cfg(any(not(feature = "rt-multi-thread"), tokio_wasi))]
+    fn default_worker_threads(_: Kind) -> Option<usize> {
+        None
+    }
+
+    #[cfg(all(feature = "rt-multi-thread", not(tokio_wasi)))]
+    fn default_worker_threads(kind: Kind) -> Option<usize> {
+        match kind {
+            // Always return None if using current thread
+            Kind::CurrentThread => return None,
+            Kind::MultiThread => {}
+        };
+        match std::env::var(ENV_WORKER_THREADS) {
+            Ok(s) => {
+                let n: usize = s.parse().unwrap_or_else(|e| {
+                    panic!(
+                        "{} must be usize, error: {}, value: {}",
+                        ENV_WORKER_THREADS, e, s
+                    )
```

Comment:
Currently we panic only if the string is valid utf-8 but not an integer, since invalid unicode hits the `Err(_)` path.

Reply:
Sorry, I didn't understand, the panic here is to check if the input is a usize.
For example, if the user inputs `TOKIO_WORKER_THREADS=3.14`, it will panic.

Reply:
Right, so `TOKIO_WORKER_THREADS=3.14` panics as it should, but `TOKIO_WORKER_THREADS=[invalid utf-8 goes here]` doesn't panic. Please see the definition of the [`VarError`](https://doc.rust-lang.org/stable/std/env/enum.VarError.html) error type:
```Rust
pub enum VarError {
    NotPresent,
    NotUnicode(OsString),
}
```
It can fail for reasons other than "the environment variable is not set".

Reply:
Thank you very much for your instruction!
This is fixed now.